odoo/tools/populate.py

```diff
@@ -0,0 +1,84 @@
+import random
+from odoo.tools import pycompat
+
+
+# format str
+# compat {** ... }
+
+def Random(seed):
+    r = random.Random()
+    r.seed(seed, version=2)
+    return r
+
+
+def format_str(val, counter, values): # todo format
+    if isinstance(val, str):
+        return val.format(counter=counter, values=values)
+    return val
+
+
+def chain_generators(fields_generates, model_name):
+    generate = root_generate()
+    for (fname, field_generate) in fields_generates:
+        generate = field_generate(generate, fname, model_name)
+    return generate
```

Comment:
Naming convention: the result of calling a `generate` should be a `generator` IMO.